specification/purview/data-plane/Azure.Analytics.Purview.Share/preview/2022-06-30-preview/share.json

```diff
@@ -0,0 +1,1796 @@
+            "description": "The tenant email registration payload",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/TenantEmailRegistration"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/TenantEmailRegistration"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/PurviewShareError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ReceivedShares_ActivateTenantEmailRegistration": {
+            "$ref": "./examples/ReceivedShares_ActivateTenantEmailRegistration.json"
+          }
+        }
+      }
+    },
+    "/registerEmail": {
+      "post": {
+        "tags": [
+          "TenantEmailRegistration"
+        ],
+        "summary": "Registers the tenant and email combination for activation.",
+        "description": "Register an email for the current tenant",
+        "operationId": "ReceivedShares_RegisterTenantEmailRegistration",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "in": "header",
+            "name": "repeatability-request-id",
+            "description": "If specified, the client directs that the request is repeatable; that is, that the client can make the request multiple times with the same Repeatability-Request-Id and get back an appropriate response without the server executing the request multiple times. The value of the Repeatability-Request-Id is an opaque string representing a client-generated, globally unique for all time, identifier for the request. It is recommended to use version 4 (random) UUIDs.",
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/TenantEmailRegistration"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "#/definitions/PurviewShareError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ReceivedShares_RegisterTenantEmailRegistration": {
+            "$ref": "./examples/ReceivedShares_RegisterTenantEmailRegistration.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ADLSGen2Artifact": {
+      "description": "An ADLS Gen2 account artifact.",
+      "required": [
+        "properties",
+        "storeKind",
+        "storeReference"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Artifact"
+        }
+      ],
+      "properties": {
+        "properties": {
```

Comment:
It would be better to "allOf" the ADLSGen2ArtifactProperties schema into this one rather than nest it inside of a "properties" property.

Reply:
Is the suggestion to drop both properties and add AdlsGen2ArtifactProperties in allOf? I don't think that would work because in the actual DTO, AdlsGen2ArtifactProperties lives under the properties of AldsGen2Artifact.

Circled in blue shows AdlsGen2ArtifactProperties